projects/plugins/social/src/js/store/actions/jetpack-settings.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * External dependencies
+ */
+/*eslint lodash/import-scope: [2, "method"]*/
+import pick from 'lodash/pick';
+import { select } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import {
+	fetchJetpackSettings,
+	updateJetpackSettings as updateJetpackSettingsControl,
+} from '../controls';
+import { STORE_ID } from '../../store';
+
+export const SET_JETPACK_SETTINGS = 'SET_JETPACK_SETTINGS';
+export const TOGGLE_PUBLICIZE_MODULE = 'TOGGLE_PUBLICIZE_MODULE';
+
+/**
+ * Yield actions to update Publicize Settings
+ *
+ * @param {object} settings - settings to apply.
+ * @yields {object} - an action object.
+ * @returns {object} - an action object.
+ */
+export function* updateJetpackSettings( settings ) {
+	try {
+		yield setUpdatingJetpackSettings();
```

Comment:
Caveat: I don't have a lot of knowledge about generator functions, but I did some reading before picking this review up. 

Is this yield statement necessary, it appears we are making a call to just update the UI state
 here in setUpdatingJetpackSettings, so I am curious why we need the yield word.

Any advantages to using a generator function as opposed to doing something like this? 

Of course, we don't have to change the existing implementation here, just wondering why the generator functions are cleaner compared to what I've proposed below. 
```
export async function updateJetpackSettings( settings) {
 try {
   dosomethingSync();
   let result = await doSomethingAsync()
   dosomethingSync();
 
   return true;
 }
 catch(e) {
   return false;
}

}
```

Reply:
It's pretty complicated and I'm definitely not an expert, but generally actions in a store are created using generators. You can read [some more about it here](https://unfoldingneurons.com/2020/wordpress-data-store-properties-action-creator-generators) (the whole series is very informative).

I don't know if using an async function instead would work the same, but I think we should stick to the default way of declaring actions as much as possible.

Reply:
Thanks for the detailed explanation, here.